text/0000-nano-contracts.md

```diff
@@ -0,0 +1,431 @@
+│ constructor(...)    │      ├──────│Nano Contract #2                          │
+│                     │      │      │Id: 00007867ff1fc65e13feb6e5aec591453d1...│
+│ make_a_bet(...)     │◀─────┤      └──────────────────────────────────────────┘
+│                     │      │      ┌──────────────────────────────────────────┐
+│ collect_prize(...)  │      ├──────┤Nano Contract #3                          │
+│                     │      │      │Id: 0000038740a7abea17b7315316779ebb686...│
+│ set_result(...)     │      │      └──────────────────────────────────────────┘
+│                     │      │      ┌──────────────────────────────────────────┐
+└─────────────────────┘      └──────┤Nano Contract #4                          │
+                                    │Id: 0000b09db96194fc3cb09f5e1afd6fca65c...│
+                                    └──────────────────────────────────────────┘
+</pre>
+
+### Interaction between Nano Contracts and Transactions
+
+There are three ways to interact with a nano contract:
+
+1. Deposit tokens into the nano contract.
+2. Withdraw tokens from the nano contract.
+3. Call a method of the nano contract.
+
+The deposit is made through a transaction with an output that calls a method of the nano contract. The method will validate whether the deposit is valid. For instance, the transaction below is depositing 1,000 HTR into the Nano Contract #1:
+
+<pre>
+┌─────────────────────────────────────────────┐
+│                Transaction A                │
+├──────────────────────┬──────────────────────┤
+│        Inputs        │       Outputs        │
+├──────────────────────┼──────────────────────┤
+│                      │  1,000 HTR           │
+│       200 HTR        │  Nano Contract #1    │
+│                      │  make_a_bet(BRA)     │
+├──────────────────────┼──────────────────────┤
+│                      │                      │
+│       900 HTR        │  100 HTR (change)    │
+│                      │                      │
+└──────────────────────┴──────────────────────┘
+</pre>
+
+The withdrawal is made through a transaction with an input that calls a method of the nano contract. The method will validate whether the widthdrawal is valid. For instance, the transaction below is withdrawing 2,500 HTR from the Nano Contract #1:
+
+<pre>
+┌────────────────────────────────────────────────────────────┐
+│                       Transaction B                        │
+├─────────────────────┬──────────────────────────────────────┤
+│       Inputs        │               Outputs                │
+├─────────────────────┼──────────────────────────────────────┤
+│  2,500 HTR          │  2,500 HTR                           │
+│  Nano Contract #1   │  P2PKH                               │
+│  collect_prize()    │  HBpGrjZcYNVJrZTy4Yct7HtUdLM1ToCPi3  │
+└─────────────────────┴──────────────────────────────────────┘
+</pre>
+
+Finally, a simple call to a method is similar to a deposit or withdrawal but with amount zero.
```

Comment:
Will we have methods that can be only used in the tx inputs and others in the outputs? Should it be explicit in the blueprint?

`collect_prize` is a method just like `make_a_bet` or any other method that can be called with amount 0. How can we know where they are allowed? Does it make sense to have a method with amount 0 called in the inputs?